src/sinks/util/tcp.rs

```diff
@@ -175,14 +175,33 @@ impl Sink for TcpSink {
 
         match self.poll_connection() {
             Ok(Async::Ready(connection)) => {
-                let mut dummy = [0; 0];
-                // Test if the remote has issued a disconnect
-                match connection.get_mut().read(&mut dummy) {
+                // This type internally is either a `TcpStream` or a
+                // `SslStream<TcpStream>`. This match will provide the inner
+                // `TcpStream`.
+                let conn = match connection.get_mut() {
+                    MaybeTls::Raw(t) => t,
+                    MaybeTls::Tls(t) => t.get_mut().get_mut(),
+                };
+
+                // Test if the remote has issued a disconnect by calling read(2)
+                // with a 0 sized buffer.
```

Comment:
I'm wondering if a zero-sized buffer is the right call. Theoretically that means we can't differentiate between an EOF `Ok(0)` and a "there was stuff to read but you gave me a zero-byte buffer to write into" `Ok(0)`. We don't actually know what to do differently in the second case, so maybe it doesn't matter?

Reply:
I agreed, I changed this to a size 1 buffer. If we do somehow get some bytes on the read side we can just ignore it.